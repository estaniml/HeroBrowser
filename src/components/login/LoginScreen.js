import { useContext } from "react";
import { useNavigate } from "react-router-dom"
import { AuthContext} from "../../auth/authContext"
import { types } from "../../types/types";

const LoginScreen = () => {
  
  const { dispatch } = useContext(AuthContext);
  const navigate = useNavigate();

  const handleLogin = () => {
    
    const action = {
      type: types.login,
      payload: { name: 'Estanislao' }
    }

    dispatch(action);

    const lastPath = localStorage.getItem('lastPath') || '/marvel';

    navigate(lastPath, {
       replace: true
    });
  }

  return (
    <div className="container mt-5">
        <h1>Iniciar Sesion</h1>
        <hr />

        <form className="form">
          <input 
              type="text"
              className="form-control"
              placeholder="username"
            />

            <input 
              type="password"
              className="form-control my-3"
              placeholder="password"
            />
            
            <button 
              className="btn btn-info text-white w-50"
              onClick={handleLogin}
            >
              Login
            </button>
        </form>
    </div>
  )
}

export default LoginScreen