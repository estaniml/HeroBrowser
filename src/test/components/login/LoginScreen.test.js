import { mount } from "enzyme";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext";
import LoginScreen from "../../../components/login/LoginScreen"
import { types } from "../../../types/types";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}));

describe('testing <LoginScreen />', () => { 

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter initialEntries={['/login']}>
                    <Routes>
                        <Route path='/login' element={<LoginScreen />}/>
                    </Routes>
                </MemoryRouter>
            </AuthContext.Provider>
    )

    test('should match with the snapshot', () => { 

        expect(wrapper).toMatchSnapshot();

    })

    test('should do dispatch and navigate', () => { 

        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: { name: 'Estanislao' }
        });

        expect(mockNavigate).toHaveBeenCalledWith('/marvel', { replace: true });

        localStorage.setItem('lastPath', '/dc')

        handleClick();

        expect( mockNavigate ).toHaveBeenCalledWith('/dc', { replace:true})
    })
})