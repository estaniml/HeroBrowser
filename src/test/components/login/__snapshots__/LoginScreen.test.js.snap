// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <LoginScreen /> should match with the snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Iniciar Sesion
  </h1>
  <hr />
  <form
    className="form"
  >
    <input
      className="form-control"
      placeholder="username"
      type="text"
    />
    <input
      className="form-control my-3"
      placeholder="password"
      type="password"
    />
    <button
      className="btn btn-info text-white w-50"
      onClick={[Function]}
    >
      Login
    </button>
  </form>
</div>
`;
